examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,247 @@
+# # Shallow Water Example: an unstable Bickley jet 
+#
+# This example shows how to use `Oceananigans.ShallowWaterModel` to simulate
+# the evolution of an unstable, geostrophically balanced, Bickley jet.
+# The model solves the governing equations for the shallow water model in
+# conservation form.  The geometry is that of a periodic channel
+# in the ``x``-direction with a flat bottom and a free-surface.  The initial
+# conditions are that of a Bickley jet with small amplitude perturbations.
+# The interested reader can see ["The nonlinear evolution of barotropically unstable jets," J. Phys. Oceanogr. (2003)](https://doi.org/10.1175/1520-0485(2003)033<2173:TNEOBU>2.0.CO;2)
+# for more details on this specific problem. 
+#
+# Unlike the other models, the fields that are simulated are the mass transports, 
+# ``uh`` and ``vh`` in the ``x`` and ``y`` directions, respectively,
+# and the height ``h``.  Note that ``u`` and ``v`` are the velocities, which 
+# can easily be computed when needed.
+#
+# ## Install dependencies
+#
+# First we make sure that we have all of the packages that are required to
+# run the simulation.
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, NCDatasets, Plots, Printf, LinearAlgebra, Polynomials"
+# ```
+#
+
+using Oceananigans
+using Oceananigans.Models: ShallowWaterModel
+
+# ## Two-dimensional domain 
+#
+# The shallow water model is a two-dimensional model and we must specify
+# the number of vertical grid points to be one.  
+# We pick the length of the domain to fit one unstable mode along the channel
+# exactly.  Note that ``Lz`` is the mean depth of the fluid.  It is determined
+# using linear stability theory that the wavenumber that yields the most unstable
+# mode is ``k=0.95``. See [Linear-Stability-Calculators](https://github.com/francispoulin/Linear-Stability-Calculators/tree/main/ShallowWater/Julia/Cartesian)
+# for details on how that is computed.
+
+Lx = 2π/0.95
+Ly = 20
+Lz = 1
+Nx = 128
+Ny = Nx
+
+grid = RegularRectilinearGrid(size = (Nx, Ny, 1),
+                            x = (0, Lx), y = (0, Ly), z = (0, Lz),
+                            topology = (Periodic, Bounded, Bounded))
+
+# ## Physical parameters
+#
+# This is a toy problem and we choose the parameters so this jet idealizes
+# a relatively narrow mesoscale jet.   
+# The physical parameters are
+#
+#   * ``f``: Coriolis parameter
+#   * ``g``: Acceleration due to gravity
+#   * ``U``: Maximum jet speed
+#   * ``\Delta\eta``: Maximum free-surface deformation that is dictated by geostrophy
+#   * ``\epsilon`` : Amplitude of the perturbation
+
+f = 1
+g = 9.80665
```

Comment:
We could also use `Oceananigans.Buoyancy.g_Earth` to get `g` but I'm not sure if that's cleaner or not... I'll leave it up to you.

Reply:
I didn't know you can do that, so thanks for pointing this out.  For this example, I think I prefer just hard coding in g but will keep this in mind for the future.  Thanks